### PR TITLE
[python] Expose MusicBrainz IDs from MusicInfoTag

### PR DESCRIPTION
**Pull request by Freso**
## Description
Adds all the MusicBrainz ID getters from the C interface’s [`MUSIC_INFO::CMusicInfoTag`](https://codedocs.xyz/xbmc/xbmc/class_m_u_s_i_c___i_n_f_o_1_1_c_music_info_tag.html) to the Python interface’s [`InfoTagMusic`](https://codedocs.xyz/xbmc/xbmc/group__python___info_tag_music.html).

**Note!** Draft PR. More to do still! Comments are still welcome though. :)

## Motivation and Context
This is a first/initial PR to make [`InfoTagMusic`](https://codedocs.xyz/xbmc/xbmc/group__python___info_tag_music.html) be feature complete relative to its C counterpart [`CMusicInfoTag`](https://codedocs.xyz/xbmc/xbmc/class_m_u_s_i_c___i_n_f_o_1_1_c_music_info_tag.html), to allow getting these information more easily in Python add‐ons. E.g., my own [`service.listenbrainz`](https://gitlab.com/Freso/service.listenbrainz) is really lacking these MBIDs to perform as wanted, making me need to explore various workarounds. (See also [`service.listenbrainz`’s issue !6](https://gitlab.com/Freso/service.listenbrainz).)

## How Has This Been Tested?

No testing has been done yet.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [x] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by Freso** [comment]
> I have added tests to cover my change

I’m not sure where these tests should be located. Help?

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Review by DaveTBlake — APPROVED** [review]
This looks find to me, and is a resasonable idea to expose the mbid data to the Python API. Sorry but I have no idea about test routines, I don't think there are any for this aspect.

File: xbmc/interfaces/legacy/InfoTagMusic.h
Comment (ronie) [inline xbmc/interfaces/legacy/InfoTagMusic.h]: please add a note to each new function that it's been added in v19
similar to:
https://github.com/xbmc/xbmc/blob/master/xbmc/interfaces/legacy/InfoTagMusic.h#L66

**Comment by phunkyfish** [comment]
@Freso do you want to progress this to merge?

**Comment by Freso** [comment]
Yes! Anything that needs changing other than @ronie’s comment? I will work on that shortly.

**Comment by phunkyfish** [comment]
I don’t believe so.

**Comment by Freso** [comment]
So… like this?

**Comment by Freso** [comment]
Oh, looks like it may need some rebasing too.

**Comment by Freso** [comment]
I don’t know what the "The milestone can't be empty" failure (under "Mergeable" is) and the Jenkins build failure seems to be unrelated to my code… ?

**Comment by phunkyfish** [comment]
The mergeable error is just cause a milestone needs to be set on the PR. We can do that after Jenkins is happy.

**Comment by phunkyfish** [comment]
jenkins build this please

**Comment by phunkyfish** [comment]
@ronie any further review comments?

**Comment by phunkyfish** [comment]
jenkins build this please

**Comment by Freso** [comment]
`TestDateTime.GetAsTimeStamp` fails on OSX-64 but passes on all other platforms… and I have no idea what I have changed here that would alter any time related behaviour… ?

**Comment by phunkyfish** [comment]
This is something to do with daylight savings time. It's unrelated and can be ignored.

Once @ronie is happy this can be merged I think.

File: xbmc/interfaces/legacy/InfoTagMusic.h
Comment (AlwinEsch) [inline xbmc/interfaces/legacy/InfoTagMusic.h]: About this better to move down a bit like on the other places. Otherwise it becomes by doxygen added as text to `@return`

Comment (AlwinEsch) [reply xbmc/interfaces/legacy/InfoTagMusic.h]: Also forgot to ask about python_v18, shouldn't that be python_v19?

File: xbmc/interfaces/legacy/InfoTagMusic.h
Comment (Freso) [reply xbmc/interfaces/legacy/InfoTagMusic.h]: Oops. Copy/paste gone wrong. Fixed now. Thanks!

File: xbmc/interfaces/legacy/InfoTagMusic.h
Comment (Freso) [reply xbmc/interfaces/legacy/InfoTagMusic.h]: Oops. Copy/paste gone wrong. Fixed now. Thanks!

I didn’t see any `@python_v19` when I did the updates earlier… but I also realised I hadn’t `fetch`’d since I last worked on it, so yes, I think _v19 is more appropriate, so I fixed those too.

**Review by AlwinEsch — APPROVED** [review]

**Comment by phunkyfish** [comment]
I’m not sure how Python API changes work. Can I assume this is ok to merge as is? Or is anything else needed?

**Comment by phunkyfish** [comment]
Jenkins build this please

**Comment by phunkyfish** [comment]
It appears that as the API bump has latest occurred for Matrix that this is ok to in. Will merge later today if there are no objections.

**Comment by phunkyfish** [comment]
jenkins build this please

**Review by enen92 — APPROVED** [review]

**Comment by phunkyfish** [comment]
Jenkins build this please

**Comment by Freso** [comment]
Oh, passing now! :tada:

**Comment by phunkyfish** [comment]
Thanks @Freso for the contribution!

**Pull request merged**
